packages/dds/merge-tree/src/endOfTreeSegment.ts

```diff
@@ -118,3 +120,90 @@ export class EndOfTreeSegment implements ISegment, IRemovalInfo {
 const notSupported = () => {
 	assert(false, 0x3ed /* operation not supported */);
 };
+
+export class StartOfTreeSegment implements ISegment, IRemovalInfo {
+	type: string = "StartOfTreeSegment";
+	readonly isEndpoint = true;
+	readonly isStartEndpoint = true;
+	constructor(private readonly mergeTree: MergeTree) {}
+	/*
+	 * segments must be of at least length one, but
+	 * removed segments will have a calculated length
+	 * of undefined/0. we leverage this to create
+	 * a 0 length segment for the start of the tree
+	 */
+	removedSeq: number = 0;
+	removedClientIds: number[] = [LocalClientId];
+	seq = 0;
+	clientId = LocalClientId;
+	cachedLength = 1;
+	isLeaf(): this is ISegment {
+		return true;
+	}
+
+	/**
+	 * the current supported usage uses the local refs to
+	 * store detached references
+	 */
+	localRefs?: LocalReferenceCollection;
+
+	/**
+	 * this segment pretends to be a sibling of the first real segment.
+	 * so compute the necessary properties to pretend to be that segment.
+	 */
+	private getStartSegProps() {
+		let firstSegment: ISegmentLeaf | undefined;
+		let depth = 1;
+		const root = this.mergeTree.root;
+		depthFirstNodeWalk(
+			root,
+			root.children[0],
+			(node) => {
+				depth++;
+				if (node?.isLeaf()) {
+					firstSegment = node;
+					return NodeAction.Exit;
+				}
+			},
+			undefined,
+			undefined,
+			false,
+		);
+		const parent = firstSegment?.parent ?? root;
+		const index = parent.childCount;
```

Comment:
this isn't a reasonable value for start of tree segment. it makes me think we probably aren't using it anywhere. Why does it need to be mocked for the end-of-tree segment?

Reply:
I would assume this value is never used for end-of-tree segment, but was added as future-proofing. Let me try changing this value to see if any tests break.